datumaro/plugins/transforms.py

```diff
@@ -1229,3 +1230,140 @@ def transform_item(self, item: DatasetItem):
                 attributes=self._filter_attrs(item.attributes), annotations=filtered_annotations
             )
         return item
+
+
+class Correct(Transform, CliPlugin):
+    """
+    Changes the content of infos.
+    A user can add meta-data of dataset such as author, comments, or related papers.
+    Infos values are not affect on the dataset structure.
+    We thus can add any meta-data freely.
+    """
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument(
+            "-r",
+            "--reports",
+            type=str,
+            default="validation_reports.json",
+            help="A validation report from a 'validate' CLI",
+        )
+        return parser
+
+    def __init__(
+        self,
+        extractor: IDataset,
+        reports: Union[str, Dict],
+    ):
+        super().__init__(extractor)
+
+        if isinstance(reports, str):
+            try:
+                # Try to load the argument as a JSON file
+                with open(reports) as file:
+                    reports = json.load(file)
+            except FileNotFoundError:
+                raise Exception("Invalid validation reports with json format")
+
+        self._reports = reports["validation_reports"]
+
+        self._remove_items = []
+        self._remove_anns = []
+        self._add_attrs = []
+
+        self._analyze_reports(report=self._reports)
+
+    def _parse_ann_ids(self, desc: str):
+        return [int(s) for s in str.split(desc, "'") if s.isdigit()][0]
+
+    def _analyze_reports(self, report):
+        for rep in report:
+            if rep["anomaly_type"] == "MissingLabelCategories":
+                label_categories = LabelCategories()
+                for item in self._extractor:
+                    for ann in item.annotations:
```

Comment:
```suggestion
                unique_labels = sorted(list({ann.label for item in self._extractor for ann in item.annotations}))
                label_categories = LabelCategories().from_iterable([f"label_{label}" for label in unique_labels])
                for item in self._extractor:
                    for ann in item.annotations:
```

For better performance we can build a set of unique labels (integer) first, rather than calling string comparator (from `label_categories.find(str(ann.label))[0]`) during the loop

Reply:
I updated. Thanks.